ra/ra.go

```diff
@@ -2084,35 +2084,39 @@ func (ra *RegistrationAuthorityImpl) RevokeCertByKey(ctx context.Context, req *r
 		return nil, err
 	}
 
-	// Perform an Akamai cache purge to handle occurrences of a client
-	// previously successfully revoking a certificate, but their cache purge had
-	// unexpectedly failed. Clients can re-attempt revocation and purge the
-	// Akamai cache.
-	if errors.Is(revokeErr, berrors.AlreadyRevoked) {
+	// If the revocation and blocked keys list addition were successful, then
+	// just purge and return.
+	if revokeErr == nil && err == nil {
```

Comment:
We already know `err == nil`, because we're past the `if err != nil` check. I think this case can be inserted between lines 2096 and 2097 below, with `if err != nil` becoming an `} else {`

Reply:
We're past the `if err != nil` check for adding the key to the blocked keys list. I can remove that portion of the check.

Reply:
I think I get what you're saying.